test/fuzz/test_runner.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Run fuzz test targets.
+"""
+
+import argparse
+import configparser
+import os
+import sys
+import subprocess
+import logging
+
+
+def main():
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        "-l",
+        "--loglevel",
+        dest="loglevel",
+        default="INFO",
+        help="log events at this level and higher to the console. Can be set to DEBUG, INFO, WARNING, ERROR or CRITICAL. Passing --loglevel DEBUG will output all logs to console.",
+    )
+    parser.add_argument(
+        '--export_coverage',
+        action='store_true',
+        help='If true, export coverage information to files in the seed corpus',
+    )
+    parser.add_argument(
+        'corpi',
+        metavar='dir_corpus',
+        nargs='+',
+        help='The seed corpus to run on (must contain subfolders for each fuzz target). Creates a coverage report when one corpus is given, otherwise merges any further corpi.',
+    )
+    args = parser.parse_args()
+
+    # Set up logging
+    logging.basicConfig(
+        format='%(message)s',
+        level=int(args.loglevel) if args.loglevel.isdigit() else args.loglevel.upper(),
+    )
+
+    # Read config generated by configure.
+    config = configparser.ConfigParser()
+    configfile = os.path.abspath(os.path.dirname(__file__)) + "/../config.ini"
+    config.read_file(open(configfile, encoding="utf8"))
+
+    # Build list of tests
+    with open(os.path.join(config["environment"]["SRCDIR"], 'src', 'Makefile.test.include'), encoding='utf-8') as makefile_test:
+        test_list = []
+        read_targets = False
+        for line in makefile_test.readlines():
+            line = line.strip().replace('test/fuzz/', '').replace(' \\', '')
+            if read_targets:
+                if not line:
+                    break
+                test_list.append(line)
+                continue
+
+            if line == 'FUZZ_TARGETS =':
+                read_targets = True
+
+    if not test_list:
+        logging.error("No fuzz targets found")
+        sys.exit(1)
+
+    logging.info("Fuzz targets found: {}".format(test_list))
+
+    if not config["components"].getboolean("ENABLE_FUZZ"):
+        logging.error("Must have fuzz targets built")
+        sys.exit(1)
+
+    help_output = subprocess.run(
+        args=[
+            os.path.join(config["environment"]["BUILDDIR"], 'src', 'test', 'fuzz', test_list[0]),
+            '-help=1',
+        ],
+        check=True,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+    ).stderr
+    if not "libFuzzer" in help_output:
```

Comment:
PEP-8 tells us to do membership tests using `if not in` instead of `if not .. in` :-)